api/api.go

```diff
@@ -418,6 +427,8 @@ func (b *BlockChainAPI) GetBlockByNumber(
 		return handleError[*ethTypes.Block](err, l, b.collector)
 	}
 
+	_ = b.blockByNumberCache.Add(height, apiBlock)
```

Comment:
Given that `GetBlockByHash`, does pretty much the exact same DB calls, we could possibly improve that one too.